.github/workflows/maven.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'adopt'
+        cache: maven
+    - name: Update
+      run: sudo apt-get update
+    - name: Install ffmpeg
+      run: sudo apt-get -y install ffmpeg
+    - name: Build with Maven
+      run: mvn -B package --file pom.xml
```

Comment:
Just curious... isn't the pom.xml the default file? Why use the `--file`?

Reply:
This is how the build file was generated by GitHub... I assumed the writer of the script generator knew what they were doing. Removed redundant flags from command.

Reply:
Yes, much of GitHub's Action documentation also uses tags (which is crazy, IMHO); they do have a security best practices page (https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) but for some reason they don't follow their own recommendations in what they present to us everywhere else.

For finding hashes: you can either copy the hashes from one of our other projects (e.g. https://github.com/UCLALibrary/hauth/blob/dca9d2866cecb9ed4bda18734c30c81b0e3bf037/.github/workflows/build.yml#L23) or, in the case where you're using an Action we've not used before, you can look up that Action on the GitHub Action's page (https://github.com/marketplace?type=actions).

Once you find an Action (e.g. https://github.com/marketplace/actions/checkout) you'll see the link to its GitHub repo on the lower right. You can follow that to the GitHub repo and look at the latest tag (e.g. https://github.com/actions/checkout/tags) and you'll see the hash for that tag (e.g. https://github.com/actions/checkout/commit/ec3a7ce113134d7a93b817d10a8272cb61118579). You can use that hash in the Action and leave a comment as to which tag it's associated with (check one of our other Actions for an example of that).